Modules/GLO/src/CTFSizeTask.cxx

```diff
@@ -0,0 +1,129 @@
+  mDefaultBinning[DetID::ITS] = "1000, 1e2, 1e5";
+  mDefaultBinning[DetID::TPC] = "1000, 1e3, 1e6";
+  mDefaultBinning[DetID::TRD] = "1000, 1e2, 1e5";
+  mDefaultBinning[DetID::TOF] = "1000, 10, 1e4";
+  mDefaultBinning[DetID::PHS] = "100, 10, 1e3";
+  mDefaultBinning[DetID::CPV] = "100, 10, 3e4";
+  mDefaultBinning[DetID::EMC] = "1000, 100, 5e5";
+  mDefaultBinning[DetID::HMP] = "100, 1, 300";
+  mDefaultBinning[DetID::MFT] = "1000, 1e2, 1e4";
+  mDefaultBinning[DetID::MCH] = "100, 1e3, 5e4";
+  mDefaultBinning[DetID::MID] = "100, 10, 500";
+  mDefaultBinning[DetID::ZDC] = "100, 1e3, 1e4";
+  mDefaultBinning[DetID::FT0] = "100, 1, 500";
+  mDefaultBinning[DetID::FV0] = "100, 1, 400";
+  mDefaultBinning[DetID::FDD] = "100, 1, 100";
+  mDefaultBinning[DetID::CTP] = "100, 1, 100";
+  mDefaultBinning[DetID::TST] = "1, 0, 1";
+
+  constexpr int nLogBins = 100;
+  float xBins[nLogBins + 1];
+  float xBinLogMin = 0.f;
+  float xBinLogMax = 11.f;
+  float logBinWidth = (xBinLogMax - xBinLogMin) / nLogBins;
+  for (int iBin = 0; iBin <= nLogBins; ++iBin) {
+    xBins[iBin] = TMath::Power(10, xBinLogMin + iBin * logBinWidth);
+  }
+  for (int iDet = 0; iDet < DetID::CTP + 1; ++iDet) {
+    mHistSizesLog[iDet] = new TH1F(Form("hSizeLog_%s", DetID::getName(iDet)), Form("%s CTF size per TF;Byte;counts", DetID::getName(iDet)), nLogBins, xBins);
+    getObjectsManager()->startPublishing(mHistSizesLog[iDet]);
+    getObjectsManager()->setDefaultDrawOptions(mHistSizesLog[iDet]->GetName(), "logx");
+  }
+}
+
+void CTFSize::startOfActivity(const Activity& activity)
+{
+  if (!mPublishingDone) {
+    for (int iDet = 0; iDet < DetID::CTP + 1; ++iDet) {
+      auto binning = getBinningFromConfig(iDet, activity);
+      std::string unit = (iDet == DetID::EMC || iDet == DetID::CPV) ? "B" : "kB";
+      mHistSizes[iDet] = new TH1F(Form("hSize_%s", DetID::getName(iDet)), Form("%s CTF size per TF;%s;counts", DetID::getName(iDet), unit.c_str()), std::get<0>(binning), std::get<1>(binning), std::get<2>(binning));
+      getObjectsManager()->startPublishing(mHistSizes[iDet]);
+    }
+    mPublishingDone = true;
+  }
+  reset();
+}
+
+void CTFSize::startOfCycle()
+{
+  // reset();
+}
+
+void CTFSize::monitorData(o2::framework::ProcessingContext& ctx)
+{
+  const auto sizes = ctx.inputs().get<std::array<size_t, DetID::CTP + 1>>("ctfSizes");
+  for (int iDet = 0; iDet <= DetID::CTP; ++iDet) {
+    std::cout << DetID::getName(iDet) << ": " << sizes[iDet] << std::endl;
+    float conversion = (iDet == DetID::EMC || iDet == DetID::CPV) ? 1.f : 1024.f; // EMC and CPV can sent <1kB per CTF
+    mHistSizes[iDet]->Fill(sizes[iDet] / conversion);
+    mHistSizesLog[iDet]->Fill(sizes[iDet]);
+  }
+}
+
+void CTFSize::endOfCycle()
+{
+}
+
+void CTFSize::endOfActivity(const Activity& /*activity*/)
+{
+}
+
+void CTFSize::reset()
+{
+  for (auto h : mHistSizes) {
+    h->Reset();
+  }
+  for (auto h : mHistSizesLog) {
+    h->Reset();
+  }
+}
```

Comment:
This might crash if we invoke `reset()` before `startOfActivity()`, I would add nullptr checks to be on the safe side.

This being said, in the current state the QC framework should not do that, so I will mark it as green and I would propose to merge it before the next release in any case.

Reply:
Thanks for the comment. I have added a protection. Yes, please merge it before the next release so that we can already get the individual size plots in the QCDB